// @kiwicom/orbit-components/lib/Modal/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ModalFooter", {
  enumerable: true,
  get: function get() {
    return _ModalFooter.default;
  }
});
Object.defineProperty(exports, "ModalHeader", {
  enumerable: true,
  get: function get() {
    return _ModalHeader.default;
  }
});
Object.defineProperty(exports, "ModalSection", {
  enumerable: true,
  get: function get() {
    return _ModalSection.default;
  }
});
exports.default = exports.PureModal = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

var _ButtonLink = _interopRequireWildcard(require("../ButtonLink"));

var _Close = _interopRequireDefault(require("../icons/Close"));

var _consts = require("./consts");

var _keyMaps = _interopRequireDefault(require("../common/keyMaps"));

var _mediaQuery = _interopRequireWildcard(require("../utils/mediaQuery"));

var _ModalFooter = _interopRequireWildcard(require("./ModalFooter"));

var _ModalHeader = _interopRequireWildcard(require("./ModalHeader"));

var _ModalSection = _interopRequireWildcard(require("./ModalSection"));

var _Heading = require("../Heading");

var _rtl = require("../utils/rtl");

var _transition = _interopRequireDefault(require("../utils/transition"));

var _ModalContext = require("./ModalContext");

var _consts2 = require("../utils/mediaQuery/consts");

var _randomID = _interopRequireDefault(require("../utils/randomID"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var getSizeToken = function getSizeToken() {
  return function (_ref) {
    var _tokens;

    var size = _ref.size,
        theme = _ref.theme;
    var tokens = (_tokens = {}, _defineProperty(_tokens, _consts.SIZES.SMALL, theme.orbit.widthModalSmall), _defineProperty(_tokens, _consts.SIZES.NORMAL, theme.orbit.widthModalNormal), _defineProperty(_tokens, _consts.SIZES.LARGE, theme.orbit.widthModalLarge), _tokens);
    return tokens[size];
  };
}; // media query only for IE 10+, not Edge


var onlyIE = function onlyIE(style) {
  var breakpoint = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "all";
  return (0, _styledComponents.css)(["@media ", " and (-ms-high-contrast:none),(-ms-high-contrast:active){", ";}"], breakpoint, style);
};

var ModalBody = _styledComponents.default.div.withConfig({
  displayName: "Modal__ModalBody",
  componentId: "sc-1f0srsl-0"
})(["width:100%;height:100%;position:fixed;top:0;right:0;bottom:0;left:0;z-index:", ";box-sizing:border-box;outline:none;overflow-x:hidden;background-color:rgba(0,0,0,0.5);font-family:", ";-webkit-overflow-scrolling:auto;", ";", ";"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.zIndexModalOverlay;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.fontfamily;
}, _mediaQuery.default.largeMobile((0, _styledComponents.css)(["overflow-y:auto;padding:", ";"], function (_ref4) {
  var theme = _ref4.theme;
  return theme.orbit.spaceXXLarge;
})), onlyIE((0, _styledComponents.css)(["position:-ms-page;"])));

ModalBody.defaultProps = {
  theme: _defaultTheme.default
};

var ModalWrapper = _styledComponents.default.div.withConfig({
  displayName: "Modal__ModalWrapper",
  componentId: "sc-1f0srsl-1"
})(["box-sizing:border-box;min-height:100%;display:flex;align-items:flex-start;margin:0 auto;position:fixed;width:100%;border-top-left-radius:9px;border-top-right-radius:9px;transition:", ";top:", ";", ";", ";"], (0, _transition.default)(["top"], "normal", "ease-in-out"), function (_ref5) {
  var loaded = _ref5.loaded;
  return loaded ? "32px" : "100%";
}, onlyIE((0, _styledComponents.css)(["height:1px;"])), _mediaQuery.default.largeMobile((0, _styledComponents.css)(["position:relative;top:0;max-width:", ";align-items:center;"], getSizeToken)));

ModalWrapper.defaultProps = {
  theme: _defaultTheme.default
};

var CloseContainer = _styledComponents.default.div.withConfig({
  displayName: "Modal__CloseContainer",
  componentId: "sc-1f0srsl-2"
})(["display:flex;", ";position:", ";top:", ";right:0;z-index:800;justify-content:flex-end;align-items:center;box-sizing:border-box;height:52px;width:100%;max-width:", ";box-shadow:", ";background-color:", ";border-top-left-radius:9px;border-top-right-radius:9px;transition:", ";", ";& + ", ":first-of-type{padding-top:52px;border-top:0;margin:0;}", "{margin-", ":", ";& svg{transition:", ";color:", ";}&:hover svg{color:", ";}&:active svg{color:", ";}}"], function (_ref6) {
  var scrolled = _ref6.scrolled,
      fixedClose = _ref6.fixedClose,
      theme = _ref6.theme;
  return fixedClose || scrolled ? (0, _styledComponents.css)(["position:fixed;", ";"], onlyIE((0, _styledComponents.css)(["position:-ms-page;"]), "(max-width:".concat(+(0, _mediaQuery.getBreakpointWidth)(_consts2.QUERIES.LARGEMOBILE, theme, true) - 1, "px)"))) : (0, _styledComponents.css)(["position:absolute;"]);
}, function (_ref7) {
  var scrolled = _ref7.scrolled,
      fixedClose = _ref7.fixedClose;
  return fixedClose || scrolled ? "fixed" : "absolute";
}, function (_ref8) {
  var scrolled = _ref8.scrolled,
      fixedClose = _ref8.fixedClose;
  return fixedClose || scrolled ? "32px" : "0";
}, function (_ref9) {
  var modalWidth = _ref9.modalWidth;
  return "".concat(modalWidth, "px");
}, function (_ref10) {
  var scrolled = _ref10.scrolled;
  return scrolled && "0 2px 4px 0 rgba(23, 27, 30, 0.1)";
}, function (_ref11) {
  var theme = _ref11.theme,
      scrolled = _ref11.scrolled;
  return scrolled && theme.orbit.paletteWhite;
}, (0, _transition.default)(["box-shadow", "background-color"], "fast", "ease-in-out"), _mediaQuery.default.largeMobile((0, _styledComponents.css)(["top:", ";right:", ";"], function (_ref12) {
  var scrolled = _ref12.scrolled,
      fixedClose = _ref12.fixedClose;
  return (fixedClose || scrolled) && "0";
}, function (_ref13) {
  var scrolled = _ref13.scrolled,
      fixedClose = _ref13.fixedClose;
  return (fixedClose || scrolled) && "auto";
})), _ModalSection.StyledModalSection, _ButtonLink.StyledButtonLink, _rtl.right, function (_ref14) {
  var theme = _ref14.theme;
  return theme.orbit.spaceXXSmall;
}, (0, _transition.default)(["color"], "fast", "ease-in-out"), function (_ref15) {
  var theme = _ref15.theme;
  return theme.orbit.paletteInkLight;
}, function (_ref16) {
  var theme = _ref16.theme;
  return theme.orbit.paletteInkLightHover;
}, function (_ref17) {
  var theme = _ref17.theme;
  return theme.orbit.paletteInkLightActive;
});

CloseContainer.defaultProps = {
  theme: _defaultTheme.default
};

var ModalWrapperContent = _styledComponents.default.div.withConfig({
  displayName: "Modal__ModalWrapperContent",
  componentId: "sc-1f0srsl-3"
})(["position:absolute;box-sizing:border-box;border-top-left-radius:9px;border-top-right-radius:9px;background-color:", ";font-family:", ";width:100%;max-height:calc( 100% - ", " - ", " );bottom:", ";box-shadow:", ";overflow-y:auto;overflow-x:hidden;", ";", "{top:", ";opacity:", ";visibility:", ";transition:", ";}", "};", "{margin-bottom:", ";}", ";", ";", ";"], function (_ref18) {
  var theme = _ref18.theme;
  return theme.orbit.backgroundModal;
}, function (_ref19) {
  var theme = _ref19.theme;
  return theme.orbit.fontFamily;
}, function (_ref20) {
  var theme = _ref20.theme;
  return theme.orbit.spaceXLarge;
}, function (_ref21) {
  var fixedFooter = _ref21.fixedFooter,
      footerHeight = _ref21.footerHeight;
  return "".concat(fixedFooter && !!footerHeight ? footerHeight : 0, "px");
}, function (_ref22) {
  var fixedFooter = _ref22.fixedFooter,
      footerHeight = _ref22.footerHeight;
  return "".concat(32 + (fixedFooter && !!footerHeight ? footerHeight : 0), "px");
}, function (_ref23) {
  var theme = _ref23.theme;
  return theme.orbit.boxShadowModal;
}, function (_ref24) {
  var fixedFooter = _ref24.fixedFooter,
      theme = _ref24.theme;
  return fixedFooter && (0, _styledComponents.css)(["", "{padding:", ";box-shadow:", ";position:fixed;transition:", ";}", ":last-of-type{padding-bottom:", ";margin-bottom:0;}"], _ModalFooter.StyledModalFooter, theme.orbit.spaceMedium, function (_ref25) {
    var fullyScrolled = _ref25.fullyScrolled;
    return fullyScrolled ? "inset 0 1px 0 ".concat(theme.orbit.paletteCloudNormal, ", 0 -2px 4px 0 rgba(23, 27, 30, 0)") : "inset 0 0 0 transparent, 0 -2px 4px 0 rgba(23, 27, 30, 0.1)";
  }, (0, _transition.default)(["box-shadow"], "fast", "ease-in-out"), _ModalSection.StyledModalSection, theme.orbit.spaceLarge);
}, _ModalHeader.MobileHeader, function (_ref26) {
  var scrolled = _ref26.scrolled,
      theme = _ref26.theme;
  return scrolled && theme.orbit.spaceXLarge;
}, function (_ref27) {
  var scrolled = _ref27.scrolled;
  return scrolled && "1";
}, function (_ref28) {
  var scrolled = _ref28.scrolled;
  return scrolled && "visible";
}, function (_ref29) {
  var scrolled = _ref29.scrolled,
      theme = _ref29.theme;
  return scrolled && "top ".concat(theme.orbit.durationNormal, " ease-in-out,\n    opacity ").concat(theme.orbit.durationFast, " ease-in-out,\n    visibility ").concat(theme.orbit.durationFast, " ease-in-out ").concat(theme.orbit.durationFast);
}, function (_ref30) {
  var scrolled = _ref30.scrolled;
  return scrolled && onlyIE((0, _styledComponents.css)(["", "{position:-ms-page;}"], _ModalHeader.MobileHeader));
}, _ModalHeader.StyledModalHeader, function (_ref31) {
  var hasModalSection = _ref31.hasModalSection,
      theme = _ref31.theme;
  return !hasModalSection && theme.orbit.spaceXLarge;
}, _mediaQuery.default.largeMobile((0, _styledComponents.css)(["position:relative;bottom:auto;border-radius:9px;padding-bottom:0;height:auto;overflow:visible;max-height:100%;", ":last-of-type{padding-bottom:", ";margin-bottom:", ";&::after{content:none;}}", "{margin-bottom:", ";}", "{padding:", ";max-width:", ";position:", ";box-shadow:", ";}", "{top:", ";width:", ";}"], _ModalSection.StyledModalSection, function (_ref32) {
  var theme = _ref32.theme;
  return theme.orbit.spaceXXLarge;
}, function (_ref33) {
  var fixedFooter = _ref33.fixedFooter,
      footerHeight = _ref33.footerHeight;
  return fixedFooter ? "".concat(footerHeight, "px") : "0";
}, _ModalHeader.StyledModalHeader, function (_ref34) {
  var hasModalSection = _ref34.hasModalSection,
      fixedFooter = _ref34.fixedFooter,
      footerHeight = _ref34.footerHeight;
  return !hasModalSection && fixedFooter ? "".concat(footerHeight, "px") : "0";
}, _ModalFooter.StyledModalFooter, function (_ref35) {
  var theme = _ref35.theme,
      fixedFooter = _ref35.fixedFooter;
  return fixedFooter ? "".concat(theme.orbit.spaceXLarge, " ").concat(theme.orbit.spaceXXLarge, "!important") : theme.orbit.spaceXXLarge;
}, function (_ref36) {
  var modalWidth = _ref36.modalWidth;
  return "".concat(modalWidth, "px");
}, function (_ref37) {
  var fullyScrolled = _ref37.fullyScrolled,
      fixedFooter = _ref37.fixedFooter;
  return fixedFooter && fullyScrolled && "absolute";
}, function (_ref38) {
  var fullyScrolled = _ref38.fullyScrolled;
  return fullyScrolled && "none";
}, _ModalHeader.MobileHeader, function (_ref39) {
  var scrolled = _ref39.scrolled,
      theme = _ref39.theme;
  return scrolled ? "0" : "-".concat(theme.orbit.spaceXXLarge);
}, function (_ref40) {
  var modalWidth = _ref40.modalWidth,
      theme = _ref40.theme;
  return "calc(".concat(modalWidth, "px - 48px - ").concat(theme.orbit.spaceXXLarge, ")");
})), onlyIE((0, _styledComponents.css)(["", "{position:", ";}"], _ModalFooter.StyledModalFooter, function (_ref41) {
  var fixedFooter = _ref41.fixedFooter;
  return fixedFooter && "-ms-page";
})), function (_ref42) {
  var theme = _ref42.theme;
  return onlyIE((0, _styledComponents.css)(["", "{position:", ";box-shadow:", ";}", ";"], _ModalFooter.StyledModalFooter, function (_ref43) {
    var fullyScrolled = _ref43.fullyScrolled,
        fixedFooter = _ref43.fixedFooter;
    return fullyScrolled && fixedFooter && "static" || fixedFooter && "fixed";
  }, function (_ref44) {
    var fixedFooter = _ref44.fixedFooter;
    return !fixedFooter && "inset 0 0 0 1px ".concat(theme.orbit.paletteWhite);
  }, function (_ref45) {
    var fullyScrolled = _ref45.fullyScrolled,
        fixedFooter = _ref45.fixedFooter;
    return fullyScrolled && fixedFooter && (0, _styledComponents.css)(["", ":last-of-type{margin-bottom:0;}", "{margin-bottom:", ";}"], _ModalSection.StyledModalSection, _ModalHeader.StyledModalHeader, function (_ref46) {
      var hasModalSection = _ref46.hasModalSection;
      return !hasModalSection && "0";
    });
  }), (0, _mediaQuery.getBreakpointWidth)(_consts2.QUERIES.LARGEMOBILE, theme));
});

ModalWrapperContent.defaultProps = {
  theme: _defaultTheme.default
};

var PureModal =
/*#__PURE__*/
function (_React$PureComponent) {
  _inherits(PureModal, _React$PureComponent);

  function PureModal() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, PureModal);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(PureModal)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "state", {
      scrolled: false,
      loaded: false,
      fixedClose: false,
      fullyScrolled: false,
      modalWidth: 0,
      footerHeight: 0,
      hasModalSection: false
    });

    _defineProperty(_assertThisInitialized(_this), "modalContent", React.createRef());

    _defineProperty(_assertThisInitialized(_this), "modalBody", React.createRef());

    _defineProperty(_assertThisInitialized(_this), "closeButton", React.createRef());

    _defineProperty(_assertThisInitialized(_this), "offset", 40);

    _defineProperty(_assertThisInitialized(_this), "setScrollPosition", function (value) {
      var _window, _modalContent$current;

      var _assertThisInitialize = _assertThisInitialized(_this),
          modalContent = _assertThisInitialize.modalContent,
          modalBody = _assertThisInitialize.modalBody;

      if (((_window = window) === null || _window === void 0 ? void 0 : _window.innerWidth) >= (0, _mediaQuery.getBreakpointWidth)(_consts2.QUERIES.LARGEMOBILE, _this.props.theme, true)) {
        var _modalBody$current;

        if (modalBody === null || modalBody === void 0 ? void 0 : (_modalBody$current = modalBody.current) === null || _modalBody$current === void 0 ? void 0 : _modalBody$current.scrollTop) {
          modalBody.current.scrollTop = value;
        }
      } else if (modalContent === null || modalContent === void 0 ? void 0 : (_modalContent$current = modalContent.current) === null || _modalContent$current === void 0 ? void 0 : _modalContent$current.scrollTop) {
        modalContent.current.scrollTop = value;
      }
    });

    _defineProperty(_assertThisInitialized(_this), "setDimensions", function () {
      var content = _this.modalContent.current;

      if (content) {
        // added in 4.0.3, interpolation of styled component return static className
        var footerEl = content.querySelector("".concat(_ModalFooter.StyledModalFooter));
        var headingEl = content.querySelector("".concat(_Heading.StyledHeading));
        _this.offset = (headingEl === null || headingEl === void 0 ? void 0 : headingEl.clientHeight) + (headingEl === null || headingEl === void 0 ? void 0 : headingEl.offsetTop);
        var contentDimensions = content.getBoundingClientRect();
        var modalWidth = contentDimensions.width;
        var footerHeight = footerEl === null || footerEl === void 0 ? void 0 : footerEl.clientHeight;

        _this.setState({
          modalWidth: modalWidth,
          footerHeight: footerHeight
        });
      }
    });

    _defineProperty(_assertThisInitialized(_this), "setHasModalSection", function () {
      if (!_this.state.hasModalSection) {
        _this.setState({
          hasModalSection: true
        });
      }
    });

    _defineProperty(_assertThisInitialized(_this), "removeHasModalSection", function () {
      if (_this.state.hasModalSection) _this.setState({
        hasModalSection: false
      });
    });

    _defineProperty(_assertThisInitialized(_this), "decideFixedFooter", function () {
      // if the content height is smaller than window height, we need to explicitly set fullyScrolled to true
      var content = _this.modalContent.current;
      var body = _this.modalBody.current; // when scrollHeight + topPadding - scrollingElementHeight is smaller or even than window height

      var fullyScrolled = (content === null || content === void 0 ? void 0 : content.scrollHeight) + 40 - (body === null || body === void 0 ? void 0 : body.scrollTop) <= window.innerHeight;

      _this.setState({
        fullyScrolled: fullyScrolled
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleResize", function () {
      _this.setDimensions();

      _this.decideFixedFooter();
    });

    _defineProperty(_assertThisInitialized(_this), "resolveAndSetStates", function (target, fullScrollOffset, fixCloseOffset) {
      _this.setState({
        scrolled: target.scrollTop >= _this.offset,
        fixedClose: target.scrollTop >= fixCloseOffset,
        fullyScrolled: _this.props.fixedFooter && // set fullyScrolled state sooner than the exact end of the scroll (with fullScrollOffset value)
        target.scrollTop >= target.scrollHeight - target.clientHeight - fullScrollOffset
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleMobileScroll", function (ev) {
      if (ev.target instanceof HTMLDivElement && ev.target === _this.modalContent.current) {
        _this.resolveAndSetStates(ev.target, 10, 1);
      }
    });

    _defineProperty(_assertThisInitialized(_this), "handleScroll", function (ev) {
      if (ev.target instanceof HTMLDivElement && ev.target === _this.modalBody.current) {
        _this.resolveAndSetStates(ev.target, 40, 40);
      }
    });

    _defineProperty(_assertThisInitialized(_this), "handleKeyDown", function (ev) {
      var onClose = _this.props.onClose;

      if (onClose && ev.key === "Escape") {
        ev.stopPropagation();
        onClose(ev);
      }

      _this.keyboardHandler(ev);
    });

    _defineProperty(_assertThisInitialized(_this), "handleClickOutside", function (ev) {
      var _this$modalContent;

      var onClose = _this.props.onClose;

      if (onClose && ((_this$modalContent = _this.modalContent) === null || _this$modalContent === void 0 ? void 0 : _this$modalContent.current) && ev.target instanceof Node && !_this.modalContent.current.contains(ev.target)) {
        // If is clicked outside of modal
        onClose(ev);
      }
    });

    _defineProperty(_assertThisInitialized(_this), "keyboardHandler", function (e) {
      if (e.keyCode === _keyMaps.default.TAB) {
        // Rotate Focus
        if (e.shiftKey && (document.activeElement === _this.firstFocusableEl || document.activeElement === _this.modalBody.current)) {
          e.preventDefault();

          _this.lastFocusableEl.focus();
        } else if (!e.shiftKey && document.activeElement === _this.lastFocusableEl) {
          e.preventDefault();

          _this.firstFocusableEl.focus();
        }
      }
    });

    _defineProperty(_assertThisInitialized(_this), "firstFocusableEl", void 0);

    _defineProperty(_assertThisInitialized(_this), "lastFocusableEl", void 0);

    _defineProperty(_assertThisInitialized(_this), "timeout", void 0);

    _defineProperty(_assertThisInitialized(_this), "modalID", void 0);

    return _this;
  }

  _createClass(PureModal, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this2 = this;

      this.timeout = setTimeout(function () {
        _this2.setState({
          loaded: true
        });

        _this2.decideFixedFooter();

        _this2.setDimensions();

        _this2.manageFocus();
      }, 15);
      this.modalID = (0, _randomID.default)("modal-");
      window.addEventListener("resize", this.handleResize);
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      if (this.props.children !== prevProps.children) {
        this.decideFixedFooter();
        this.setDimensions();
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      window.removeEventListener("resize", this.handleResize);

      if (this.timeout) {
        clearTimeout(this.timeout);
      }
    }
  }, {
    key: "manageFocus",
    value: function manageFocus() {
      var focusableElements = this.modalContent.current.querySelectorAll(_consts.FOCUSABLE_ELEMENT_SELECTORS);

      if (focusableElements.length > 0) {
        var firstFocusableEl = focusableElements[0];
        var lastFocusableEl = focusableElements[focusableElements.length - 1];
        this.firstFocusableEl = firstFocusableEl;
        this.lastFocusableEl = lastFocusableEl;
      }

      if (this.modalBody.current) this.modalBody.current.focus();
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          onClose = _this$props.onClose,
          children = _this$props.children,
          _this$props$size = _this$props.size,
          size = _this$props$size === void 0 ? _consts.SIZES.NORMAL : _this$props$size,
          _this$props$fixedFoot = _this$props.fixedFooter,
          fixedFooter = _this$props$fixedFoot === void 0 ? false : _this$props$fixedFoot,
          dataTest = _this$props.dataTest;
      var _this$state = this.state,
          scrolled = _this$state.scrolled,
          loaded = _this$state.loaded,
          fixedClose = _this$state.fixedClose,
          fullyScrolled = _this$state.fullyScrolled,
          modalWidth = _this$state.modalWidth,
          footerHeight = _this$state.footerHeight,
          hasModalSection = _this$state.hasModalSection;
      return React.createElement(ModalBody, {
        tabIndex: "0",
        onKeyDown: this.handleKeyDown,
        onScroll: this.handleScroll,
        onClick: this.handleClickOutside,
        "data-test": dataTest,
        ref: this.modalBody,
        role: "dialog",
        "aria-modal": "true",
        "aria-labelledby": this.modalID
      }, React.createElement(ModalWrapper, {
        size: size,
        loaded: loaded,
        onScroll: this.handleMobileScroll,
        fixedFooter: fixedFooter,
        id: this.modalID
      }, React.createElement(ModalWrapperContent, {
        size: size,
        fixedFooter: fixedFooter,
        scrolled: scrolled,
        ref: this.modalContent,
        fixedClose: fixedClose,
        fullyScrolled: fullyScrolled,
        modalWidth: modalWidth,
        footerHeight: footerHeight,
        hasModalSection: hasModalSection
      }, React.createElement(CloseContainer, {
        modalWidth: modalWidth,
        scrolled: scrolled,
        fixedClose: fixedClose
      }, onClose && React.createElement(_ButtonLink.default, {
        onClick: onClose,
        size: "normal",
        icon: React.createElement(_Close.default, null),
        transparent: true,
        dataTest: _consts.CLOSE_BUTTON_DATA_TEST,
        ref: this.closeButton
      })), React.createElement(_ModalContext.ModalContext.Provider, {
        value: {
          setDimensions: this.setDimensions,
          decideFixedFooter: this.decideFixedFooter,
          setHasModalSection: this.setHasModalSection,
          removeHasModalSection: this.removeHasModalSection,
          hasModalSection: hasModalSection
        }
      }, children))));
    }
  }]);

  return PureModal;
}(React.PureComponent);

exports.PureModal = PureModal;

_defineProperty(PureModal, "defaultProps", {
  theme: _defaultTheme.default
});

var ThemedModal = (0, _styledComponents.withTheme)(PureModal);
ThemedModal.displayName = "Modal";
var _default = ThemedModal;
exports.default = _default;
