// @kiwicom/orbit-components/lib/_deprecated/Card/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var _Close = _interopRequireDefault(require("../../icons/Close"));

var _ButtonLink = _interopRequireDefault(require("../../ButtonLink"));

var _CardSection = require("./CardSection");

var _CardHeader = require("./CardHeader");

var _CardContent = require("./CardContent");

var _getSpacingToken = _interopRequireDefault(require("../../common/getSpacingToken"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var StyledCard = _styledComponents.default.div.withConfig({
  displayName: "Card__StyledCard",
  componentId: "vnl6dw-0"
})(["width:100%;height:100%;background:", ";border-radius:", ";border-width:", ";border-style:", ";border-color:", ";box-sizing:border-box;position:relative;margin-bottom:", ";", "{padding-right:", ";}", " + ", ",", " + ", "{padding-top:0;border-top:0;}"], function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.backgroundCard;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.borderRadiusNormal;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.borderWidthCard;
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.orbit.borderStyleCard;
}, function (_ref5) {
  var theme = _ref5.theme;
  return theme.orbit.borderColorCard;
}, _getSpacingToken.default, _CardHeader.StyledCardHeader, function (_ref6) {
  var theme = _ref6.theme,
      closable = _ref6.closable;
  return closable && theme.orbit.spaceLarge;
}, _CardHeader.StyledCardHeader, _CardSection.StyledCardSection, _CardHeader.StyledCardHeader, _CardContent.StyledCardContent);

StyledCard.defaultProps = {
  theme: _defaultTheme.default
};

var CloseContainer = _styledComponents.default.div.withConfig({
  displayName: "Card__CloseContainer",
  componentId: "vnl6dw-1"
})(["position:absolute;top:0;right:0;z-index:1;"]);

var Card = function Card(_ref7) {
  var closable = _ref7.closable,
      onClose = _ref7.onClose,
      children = _ref7.children,
      dataTest = _ref7.dataTest,
      spaceAfter = _ref7.spaceAfter;
  return React.createElement(StyledCard, {
    closable: closable,
    "data-test": dataTest,
    spaceAfter: spaceAfter
  }, closable && React.createElement(CloseContainer, null, React.createElement(_ButtonLink.default, {
    type: "secondary",
    size: "small",
    icon: React.createElement(_Close.default, null),
    onClick: onClose,
    transparent: true
  })), children);
};

var _default = Card;
exports.default = _default;
