// @kiwicom/orbit-components/es/Modal/ModalSection/index.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import * as React from "react";
import styled, { css } from "styled-components";
import defaultTheme from "../../defaultTheme";
import media from "../../utils/mediaQuery";
import { StyledModalFooter } from "../ModalFooter";
import { withModalContext } from "../ModalContext";
export const StyledModalSection = styled.section.withConfig({
  displayName: "ModalSection__StyledModalSection",
  componentId: "sc-1ayrdn8-0"
})(["width:100%;padding:", ";background-color:", ";border-bottom:1px solid ", ";box-sizing:border-box;&:first-of-type{border-top:", ";border-top-left-radius:9px;border-top-right-radius:9px;margin-top:", ";}&:last-of-type{border-bottom:", ";border-bottom-left-radius:9px;border-bottom-right-radius:9px;& ~ ", "{margin-top:", ";}&:not(:last-child){border-radius:0;}}", ";"], ({
  theme
}) => `${theme.orbit.spaceLarge} ${theme.orbit.spaceMedium}`, ({
  theme,
  suppressed
}) => suppressed ? theme.orbit.paletteCloudLight : theme.orbit.paletteWhite, ({
  theme
}) => theme.orbit.paletteCloudNormal, ({
  suppressed,
  theme
}) => suppressed && `1px solid ${theme.orbit.paletteCloudNormal}`, ({
  suppressed,
  theme
}) => suppressed && theme.orbit.spaceLarge, ({
  suppressed,
  theme
}) => suppressed ? `1px solid ${theme.orbit.paletteCloudNormal}` : "0", StyledModalFooter, ({
  theme,
  suppressed
}) => suppressed && theme.orbit.spaceMedium, media.largeMobile(css(["padding:", ";&:first-of-type{margin-top:", ";}&:last-of-type{& ~ ", "{padding-top:", ";margin-top:0;}}"], ({
  theme
}) => theme.orbit.spaceXXLarge, ({
  suppressed,
  theme
}) => suppressed && theme.orbit.spaceXXLarge, StyledModalFooter, ({
  suppressed
}) => !suppressed && "0")));
StyledModalSection.defaultProps = {
  theme: defaultTheme
};

class ModalSection extends React.PureComponent {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "callContextFunctions", () => {
      const {
        setDimensions,
        decideFixedFooter
      } = this.props;

      if (setDimensions) {
        setDimensions();
      }

      if (decideFixedFooter) {
        decideFixedFooter();
      }
    });
  }

  componentDidMount() {
    const {
      setHasModalSection
    } = this.props;
    this.callContextFunctions();

    if (setHasModalSection) {
      setHasModalSection();
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps !== this.props) {
      this.callContextFunctions();
      const {
        setHasModalSection
      } = this.props;

      if (setHasModalSection) {
        setHasModalSection();
      }
    }
  }

  componentWillUnmount() {
    const {
      removeHasModalSection
    } = this.props;
    this.callContextFunctions();

    if (removeHasModalSection) {
      removeHasModalSection();
    }
  }

  render() {
    const {
      suppressed,
      children,
      dataTest
    } = this.props;
    return React.createElement(StyledModalSection, {
      suppressed: suppressed,
      "data-test": dataTest
    }, children);
  }

}

const DecoratedComponent = withModalContext(ModalSection);
DecoratedComponent.displayName = "ModalSection";
export default DecoratedComponent;
