// @kiwicom/orbit-components/es/_deprecated/Card/CardHeader/index.js
import * as React from "react";
import styled from "styled-components";
import defaultTheme from "../../../defaultTheme";
import Heading from "../../../Heading";
import Text from "../../../Text";
export const StyledCardHeader = styled.div.withConfig({
  displayName: "CardHeader__StyledCardHeader",
  componentId: "sc-39t2tz-0"
})(["position:relative;width:100%;padding:", ";box-sizing:border-box;color:", ";"], ({
  theme
}) => theme.orbit.spaceLarge, ({
  theme
}) => theme.orbit.colorHeading);
StyledCardHeader.defaultProps = {
  theme: defaultTheme
};
const StyledHeadingWrapper = styled.div.withConfig({
  displayName: "CardHeader__StyledHeadingWrapper",
  componentId: "sc-39t2tz-1"
})(["display:flex;"]);
const StyledSubTitle = styled.div.withConfig({
  displayName: "CardHeader__StyledSubTitle",
  componentId: "sc-39t2tz-2"
})(["font-family:", ";text-align:center;margin-top:", ";"], ({
  theme
}) => theme.orbit.fontFamily, ({
  theme
}) => theme.orbit.spaceXSmall);
StyledSubTitle.defaultProps = {
  theme: defaultTheme
};
const StyledIcon = styled.div.withConfig({
  displayName: "CardHeader__StyledIcon",
  componentId: "sc-39t2tz-3"
})(["color:", ";display:flex;align-items:center;margin-right:", ";"], ({
  theme
}) => theme.orbit.colorHeading, ({
  theme
}) => theme.orbit.spaceXSmall);
StyledIcon.defaultProps = {
  theme: defaultTheme
};

const CardHeader = ({
  icon,
  title,
  subTitle
}) => React.createElement(StyledCardHeader, null, React.createElement(StyledHeadingWrapper, null, icon && React.createElement(StyledIcon, null, icon), React.createElement(Heading, {
  type: "title2",
  element: "h2"
}, title)), subTitle && React.createElement(StyledSubTitle, null, React.createElement(Text, null, subTitle)));

export default CardHeader;
