// @kiwicom/orbit-components/es/Card/CardHeader/index.js
import * as React from "react";
import styled from "styled-components";
import defaultTheme from "../../defaultTheme";
import Heading, { StyledHeading } from "../../Heading";
import Text from "../../Text";
import { rtlSpacing } from "../../utils/rtl";
export const StyledCardHeader = styled.div.withConfig({
  displayName: "CardHeader__StyledCardHeader",
  componentId: "lvjkay-0"
})(["position:relative;width:100%;padding:", ";box-sizing:border-box;color:", ";"], ({
  theme
}) => theme.orbit.spaceLarge, ({
  theme
}) => theme.orbit.colorHeading);
StyledCardHeader.defaultProps = {
  theme: defaultTheme
};
const StyledHeadingWrapper = styled.div.withConfig({
  displayName: "CardHeader__StyledHeadingWrapper",
  componentId: "lvjkay-1"
})(["display:flex;align-items:start;", "{width:100%;}"], StyledHeading);
const StyledSubTitle = styled.div.withConfig({
  displayName: "CardHeader__StyledSubTitle",
  componentId: "lvjkay-2"
})(["display:flex;margin-top:", ";"], ({
  theme
}) => theme.orbit.spaceXSmall);
StyledSubTitle.defaultProps = {
  theme: defaultTheme
};
const StyledIcon = styled.div.withConfig({
  displayName: "CardHeader__StyledIcon",
  componentId: "lvjkay-3"
})(["color:", ";display:flex;align-items:center;margin:", ";"], ({
  theme
}) => theme.orbit.colorHeading, ({
  theme
}) => rtlSpacing(`0 ${theme.orbit.spaceXSmall} 0 0`));
StyledIcon.defaultProps = {
  theme: defaultTheme
};

const CardHeader = ({
  icon,
  title,
  subTitle,
  actions,
  dataTest
}) => React.createElement(StyledCardHeader, {
  "data-test": dataTest
}, React.createElement(StyledHeadingWrapper, null, icon && React.createElement(StyledIcon, null, icon), React.createElement(Heading, {
  type: "title2",
  element: "h2"
}, title), actions), subTitle && React.createElement(StyledSubTitle, null, React.createElement(Text, null, subTitle)));

CardHeader.displayName = "CardHeader";
export default CardHeader;
